### PR TITLE
Throw an exception in CompareInfo:IndexOf() when CompareOptions.IgnoreCase is used and managed collation is disabled.

### PR DESCRIPTION
**Pull request by vargaz**
Re: https://github.com/mono/mono/issues/19034.

### CODE REVIEW

File: mcs/class/corlib/ReferenceSources/CompareInfo.cs [hunk 1]
```diff
@@ -123,9 +123,13 @@ int internal_index_switch (string s1, int sindex, int count, string s2, CompareO
...
 				return first ? s1.IndexOfUnchecked (s2, sindex, count) : s1.LastIndexOfUnchecked (s2, sindex, count);
 			
-			return UseManagedCollation ?
-				internal_index_managed (s1, sindex, count, s2, opt, first) :
-				internal_index (s1, sindex, count, s2, first);
+			if (UseManagedCollation) {
+				return internal_index_managed (s1, sindex, count, s2, opt, first);
+			} else {
+				if (opt == CompareOptions.IgnoreCase)
+					throw new NotImplementedException ("CompareOptions.IgnoreCase is not supported on this platform.");
```
Comment (lewing) [inline mcs/class/corlib/ReferenceSources/CompareInfo.cs:130]:
```suggestion
					throw new PlatformNotSupportedException ("CompareOptions.IgnoreCase is not supported on this platform.");
```

**Comment by vargaz** [comment]
The wasm failures are related.

**Comment by lewing** [comment]
This change is fine but failing on wasm because it is missing the collation tables.

**Comment by vargaz** [comment]
Obsolete.

**Pull request closed without merging**